simulation/traffic_simulator/src/entity/entity_base.cpp

```diff
@@ -649,6 +649,13 @@ void EntityBase::requestSpeedChange(
   }
 }
 
+auto EntityBase::isControlledBySimulator() const -> bool
+{
+  THROW_SEMANTIC_ERROR(
+    getEntityTypename(),
+    " type entities do not support switching control - to be controlled by simulator or not.");
```

Comment:
I've considered this approach, but I chose not to add variable `is_controlled_by_simulator_` to the base class, as it implies that **it is used for some purpose for each entity** - and as I understand correctly, it is not intended to enable/disable control of any `Entity` by the simulator (except `Ego`).

If we add the variable `is_controlled_by_simulator_` to the base class, it would be appropriate to add a check "if it is true" every time when the simulator intends to change the `Entity` state.

For this reason, I did not add the variable `is_controlled_by_simulator_` to the base class, and the `isControlledBySimulator` method called for the base class throws an exception - because this class does not allow this type of control. Due to the architecture, the method needs to have a definition in `EntityBase`, but I think it is a good idea to make sure that it is not callable for other entities (since such switching of control is not implemented for them). 
I believe that such unambiguity of purpose of this variable can be helpful.

Reply:
@dmoszynski 
Instead of moving the variable `is_controlled_by_simulator_` to the base class, what if the implementation of EntityBase::isControlledBySimulator is `return true;`? That is, it returns a literal true. EntityBase expresses that the control of entities in traffic_simulator is by default by the simulator, and EgoEntity overrides isControlledBySimulator for special control.

In other words, even if an EntityBase derived type does not support switching between control being by the simulator and by something else, it should always be able to answer whether it is by the simulator or not.

Reply:
@yamacir-kit 
To be more precise, my intention was that method X is intended only for EgoEntity, and the attempt to check for, for example, Vehicle should not happen - checking it for such entities just does not make sense, and I wanted to signal this to the others.

However, I understand your point of view as well and return true is sensible. I will make this change.